sqlglot/executor/python.py

```diff
@@ -286,23 +286,29 @@ def aggregate(self, step, context):
     def sort(self, step, context):
         projections = self.generate_tuple(step.projections)
 
-        sink = self.table(step.projections)
+        projection_columns = [p.alias_or_name for p in step.projections]
+        all_columns = list(context.columns) + projection_columns
+        sink = self.table(all_columns)
 
         for reader, ctx in context:
-            sink.append(ctx.eval_tuple(projections))
+            sink.append(reader.row + ctx.eval_tuple(projections))
 
-        context = self.context(
+        sort_ctx = self.context(
             {
                 None: sink,
                 **{table: sink for table in context.tables},
             }
         )
-        context.sort(self.generate_tuple(step.key))
+        sort_ctx.sort(self.generate_tuple(step.key))
 
         if not math.isinf(step.limit):
-            context.table.rows = context.table.rows[0 : step.limit]
+            sort_ctx.table.rows = sort_ctx.table.rows[0 : step.limit]
 
-        return self.context({step.name: context.table})
+        output = Table(
+            projection_columns,
+            rows=[r[len(context.columns) : len(all_columns)] for r in sort_ctx.table.rows],
```

Comment:
This isn't the most efficient thing to do... but I couldn't immediately see a better solution that didn't involve more fundamental changes.

Reply:
what does this do and what is the more fundamental change ?

Reply:
This copies `rows`, keeping only the projections columns.

A more fundamental change would let us do this without copying. I don't know exactly how to do that yet. `column_range` is a similar idea, but it doesn't work as-is.